examples/bri-3/src/bri/zkp/components/merkleTree/merkleTree.ts

```diff
@@ -0,0 +1,74 @@
+import { Tree } from "./tree"
+import { IMerkleTree} from "./merkleTree.interface";
+import { PreciseProofs } from "ew-precise-proofs-js";
+
+export class MerkleTree implements IMerkleTree {
+  tree: Tree;
+
+  constructor(document: any){
+    const merkleTree: Tree = this.createTree(document);
+
+    this.tree.leaves = merkleTree.leaves;
+    this.tree.root = merkleTree.root; 
+  }
+
+  private createTree(document: any): Tree {
+    try {
+      var tree: Tree;
+
+      const leaves: PreciseProofs.Leaf[] = this.createLeaves(document);
+
+      const merkleTree = PreciseProofs.createMerkleTree(
+        leaves.map((leaf: PreciseProofs.Leaf) => leaf.hash)
+      );
+
+      const root = PreciseProofs.getRootHash(merkleTree);
+      
+      tree.leaves = leaves;
+      tree.root = root;
+
+      return tree;
+    } catch (e) {
+      console.log("Tree not created");
+      console.log(e);
+    }
+  }
+
+  private createLeaves(document: any): PreciseProofs.Leaf[] {
+    try {
+      const leaves = PreciseProofs.createLeafs(document);
```

Comment:
return PreciseProofs.createLeafs(document)
in case of an error, same thing as createTree, returning undefined for leaves here might lead to some bugs, we will need either guards everywhere we use leaves or returning default here
btw, i wonder why this library is using wrong plural for leaf? :)